src/cockpit/channels/filesystem.py

```diff
@@ -170,12 +170,27 @@ def do_open(self, options):
 
     def do_data(self, data):
         if self._tempfile is None:
+            # if the file exists already and we have an expected tag, check it
```

Comment:
This logic is almost ready to be split out into a separate `create_temp_file()` function...

Reply:
OK, next time I touch it :sweat_smile: